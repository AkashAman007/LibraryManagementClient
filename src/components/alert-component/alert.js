import React from 'react';
import Alert from "react-bootstrap/Alert";

export const AlertDismissible = (props) => {
  console.log({props});
  const {alert, removeAlert} = props;
  const {visible, error, message} = alert;
  const title = error ? "Error" : "Success!!"
  if (visible) {
    return (
      <Alert  data-testid="alert-element" variant={error ? "danger": "success"} onClose={() => removeAlert()} dismissible>
        <Alert.Heading>{title}</Alert.Heading>
        <p>
          {message}
        </p>
      </Alert>
    );
  }
  return <div/>
}
