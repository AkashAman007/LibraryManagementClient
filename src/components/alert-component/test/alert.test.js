import React from 'react';
import {render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import  { AlertDismissible }  from "../alert";

afterEach(cleanup);

it("Alert Renders Error Message", () => {
    const alert = { visible: true, error: true, message: "Unavailable" };
    const removeAlert = () => {};
    const tree = render(<AlertDismissible alert={alert}  removeAlert={removeAlert}/>)
    expect(tree.asFragment()).toMatchSnapshot();
});

it("Alert Renders Success Message", () => {
    const alert = { visible: true, error: false, message: "Success" };
    const removeAlert = () => {};
    const tree = render(<AlertDismissible alert={alert}  removeAlert={removeAlert}/>)
    expect(tree.asFragment()).toMatchSnapshot();
});

it("Alert renders nothing", () => {
    const alert = { visible: false, error: true, message: "" };
    const removeAlert = () => {};
    const tree = render(<AlertDismissible alert={alert}  removeAlert={removeAlert}/>)
    expect(tree.asFragment()).toMatchSnapshot();
});

it("Have alert-danger class in case in case of error", () => {
    const alert = { visible: true, error: true, message: "" };
    const removeAlert = () => {};
    const tree = render(<AlertDismissible alert={alert}  removeAlert={removeAlert}/>)
    const alertElement = tree.getByTestId('alert-element');
    expect(alertElement).toHaveClass('alert alert-danger');
});

it("Have alert-success class in case in case of error", () => {
    const alert = { visible: true, error: false, message: "" };
    const removeAlert = () => {};
    const tree = render(<AlertDismissible alert={alert}  removeAlert={removeAlert}/>)
    const alertElement = tree.getByTestId('alert-element');
    expect(alertElement).toHaveClass('alert alert-success');
});

it("Alert Title to have Error Text in case of error", () => {
    const alert = { visible: true, error: true, message: "" };
    const removeAlert = () => {};
    const tree = render(<AlertDismissible alert={alert}  removeAlert={removeAlert}/>)
    const errorText = tree.getByText("Error");
    expect(errorText).toBeTruthy;
});

it("Alert Title to have Success Text in case of error", () => {
    const alert = { visible: true, error: false, message: "" };
    const removeAlert = () => {};
    const tree = render(<AlertDismissible alert={alert}  removeAlert={removeAlert}/>)
    const successText = tree.getByText("Success!!");
    expect(successText).toBeTruthy;
});