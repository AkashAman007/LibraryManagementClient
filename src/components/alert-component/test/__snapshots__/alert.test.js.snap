// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Renders Error Message 1`] = `
<DocumentFragment>
  <div
    class="fade alert alert-danger alert-dismissible show"
    data-testid="alert-element"
    role="alert"
  >
    <button
      class="close"
      type="button"
    >
      <span
        aria-hidden="true"
      >
        ×
      </span>
      <span
        class="sr-only"
      >
        Close alert
      </span>
    </button>
    <div
      class="alert-heading h4"
    >
      Error
    </div>
    <p>
      Unavailable
    </p>
  </div>
</DocumentFragment>
`;

exports[`Alert Renders Success Message 1`] = `
<DocumentFragment>
  <div
    class="fade alert alert-success alert-dismissible show"
    data-testid="alert-element"
    role="alert"
  >
    <button
      class="close"
      type="button"
    >
      <span
        aria-hidden="true"
      >
        ×
      </span>
      <span
        class="sr-only"
      >
        Close alert
      </span>
    </button>
    <div
      class="alert-heading h4"
    >
      Success!!
    </div>
    <p>
      Success
    </p>
  </div>
</DocumentFragment>
`;

exports[`Alert renders nothing 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
